src/main.c

```diff
@@ -84,8 +84,10 @@ int main(int argc, char** argv)
     if (!opt.outputFile) {
         opt.outputFile = strdup("%Y-%m-%d-%H%M%S_$wx$h_scrot.png");
         opt.thumbFile = strdup("%Y-%m-%d-%H%M%S_$wx$h_scrot-thumb.png");
-    } else
+    } else {
+        opt.thumbFile = nameThumbnail(opt.outputFile);
```

Comment:
Even if the `-t` option is not used, it is always called, it should be:
```C
    } else {
        if (opt.thumb)
            opt.thumbFile = optionsNameThumbnail(opt.outputFile);
        scrotHaveFileExtension(opt.outputFile, &haveExtension);
    }
```

Reply:
Can you please ellaborate?
From what I am understanding, you would like me to put the naming of the thumbFile in an if that's checking if opt.thumb exists? Can there be such a case that it doesn't exist there at the time of calling?

I did this time test the changes with this: ```scrot -t 50 && scrot -t 50 && scrot -t 50 file && scrot -t 50 file && scrot -t 50 file.png && scrot -t 50 file.png && scrot -t 50 file.jpg && scrot -t 50 file.jpg && scrot -t 50 file.jpeg && scrot -t 50 file.jpeg && scrot -t 50 -F file && scrot -t 50 -F file && scrot -t 50 -F file.png && scrot -t 50 -F file.png && scrot -t 50 -F file.jpg && scrot -t 50 -F file.jpg && scrot -t 50 -F file.jpeg && scrot -t 50 -F file.jpeg && scrot -F file -t 50 && scrot -F file -t 50 && scrot -F file.png -t 50 && scrot -F file.png -t 50 && scrot -F file.jpg -t 50 && scrot -F file.jpg -t 50 && scrot -F file.jpeg -t 50 && scrot -F file.jpeg -t 50``` and never ran into a situation that it doesn't exist.

Shouldn't the check if opt.thumb exist be unnecessary?

Reply:
I show you first the behavior of the `master` branch:
```gdb
(gdb) b nameThumbnail
Breakpoint 1 at 0x535e: file options.c, line 358.
(gdb) r file.png
[Inferior 1 (process 3032) exited normally]
```
As you can see if the `-t` option is not used the function is never called, which is what is expected.

Now your branch:

```gdb
(gdb) b nameThumbnail
Breakpoint 1 at 0x5361: file options.c, line 355.
(gdb) r file.png

Breakpoint 1, nameThumbnail (name=0x5555555612a0 "file.png") at options.c:355
355     {

```

The function is called even if the `-t` option is not used, and this changes the behavior that it has so far.

Reply:
Ah! Thank you! I get what you mean now. You weren't saying that calling the function if there is no -t option will cause the software to throw an error. You don't want the option to be called when there's no need. That makes sense.